ui/app/AppLayouts/stores/RootStore.qml

```diff
@@ -121,147 +391,197 @@ QtObject {
-        communitiesModuleInst.cancelRequestToJoinCommunity(id)
-    }
 
     property string currentCurrency: walletSection.currentCurrency
 
     function getFiatValue(balance, cryptoSymbol, fiatSymbol) {
         return profileSectionStore.ensUsernamesStore.getFiatValue(balance, cryptoSymbol, fiatSymbol)
     }
 
-    function setCurrentUserStatus(newStatus) {
-        if (d.userProfileInst && d.userProfileInst.currentUserStatus !== newStatus) {
-            mainModuleInst.setCurrentUserStatus(newStatus)
+    signal showToastAccountAdded(string name)
+    signal showToastAccountRemoved(string name)
+    signal showToastKeypairRenamed(string oldName, string newName)
+    signal showNetworkEndpointUpdated(string name, bool isTest)
+    signal showToastKeypairRemoved(string keypairName)
+    signal showToastKeypairsImported(string keypairName, int keypairsCount, string error)
+    signal showTransactionToast(string uuid,
+                                int txType,
+                                int fromChainId,
+                                int toChainId,
+                                string fromAddr,
+                                string fromName,
+                                string toAddr,
+                                string toName,
+                                string txToAddr,
+                                string txToName,
+                                string txHash,
+                                bool approvalTx,
+                                string fromAmount,
+                                string toAmount,
+                                string fromAsset,
+                                string toAsset,
+                                string username,
+                                string publicKey,
+                                string packId,
+                                string communityId,
+                                string communityName,
+                                int communityInvolvedTokens,
+                                string communityTotalAmount,
+                                string communityAmount1,
+                                bool communityAmountInfinite1,
+                                string communityAssetName1,
+                                int communityAssetDecimals1,
+                                string communityAmount2,
+                                bool communityAmountInfinite2,
+                                string communityAssetName2,
+                                int communityAssetDecimals2,
+                                string communityInvolvedAddress,
+                                int communityNubmerOfInvolvedAddresses,
+                                string communityOwnerTokenName,
+                                string communityMasterTokenName,
+                                string communityDeployedTokenName,
+                                string status,
+                                string error)
```

Comment:
Wow, what a signal :D

Reply:
This is just a movement to remove `mainModule`direct access from `appMain`. I didn't go deep through the implementation of all these functions and signatures.. I agree, we should review these, but it's out of this scope!